accepted/0000-subscription-matching-in-public-clouds.md

```diff
@@ -0,0 +1,194 @@
+
+This way, the Administrators will be able to define and/or gather their virtualization environments in the public clouds and make the subscription matcher do a succesfull matching of the "1-2 Virtual Machine" subscriptions.
+
+# Detailed design
+[design]: #detailed-design
+
+Most of the implementation effort needed is about coding each one of the needed plugins for `virtual-host-gatherer`. On the Java side, we would need to adapt `VirtualHostManagerProcessor.java` and most probably also `VirtualInstanceManager.java` in a similar way that we do when processing a KUBERNETES virtual host, since we don't always know about amount of RAM, number of CPUs, etc from the Virtualization Host.
+
+
+## 1-2 Virtual Machine Licensing
+From [SUSE Terms & Conditions](https://www.suse.com/products/terms_and_conditions.pdf):
+
+> Up to 2 Virtual Machines running on the same Virtualization Host or Virtualization Environment or within the same Private Cloudaccount or **Public Cloud zone can be deployed with one "1-2 Sockets or 1-2 Virtual Machines" Subscription Offering**.
+
+This means, all virtual instances that belong to the same "zone" (not region) will match with the "1-2 Virtual Machines" subscription.
+
+## The output from a `virtual-host-gatherer` plugin:
+The required output from a plugin is a JSON response to the STDOUT, that looks like the following:
+
+```json
+{
+    "my-amazon-us-east-2c": {
+        "my-amazon-us-east-2c": {
+            "name": "my-amazon-us-east-2c",
+            "hostIdentifier": "my-amazon-us-east-2c",
+            "ramMb": 0,
+            "cpuArch": "aws",
+            "cpuMhz": 0,
+            "os": "Amazon AWS",
+            "osVersion": "Amazon AWS",
+            "totalCpuCores": 0,
+            "totalCpuSockets": 0,
+            "type": "aws",
+            "vms": {
+                "instance4": "i-fffcb2bd24b7b9",
+                "instance5": "i-fffcb2bd24b7b11"
+            }
+        }
+    }
+}
+```
+
+All the attributes on the above JSON example are currently required by SUMA to properly create the entries on the DB. Since we don't know the real hardware used to virtualize the instances, those values like `ramMb`, `cpuMhz`, `totalCpuCores`, `totalCpuSockets` have been faked to 0. As mentioned before, with some adaptations in the Java side we could not require those values (like for KUBERNETES).
+
+NOTE: All the "vms" that belongs to the same "Virtual Host Manager", are considered in the same "Virtualization group" for subscription matching. In the case of Public Clouds, this means a different "Virtual Host Manager" needs to be added for each one of the "Zones" of the particular Public Cloud provider where we want gather virtual instances.
+
+## Add new Virtual Instance Types to the DB
+New virtual instance types would be needed on the `rhnVirtualInstanceType` database table, these would be `azure`, `aws`, `gce` and `generic` to indicate the type of the instance.
```

Comment:
As you say below, there is already the file based type which should be used as `generic`. That type was always meant to be generic, but until now we only have VMware so the example data reflect VMware. 

I would remove `generic` here and explain more on top, that the existing file based type can be used for all types of virtual host managers and that we only need to adapt the docs.